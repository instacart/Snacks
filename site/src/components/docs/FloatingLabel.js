import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

The floating label must be used inside of a TextField or component that is 'position: relative'. See TextField to see how FloatingLabel is used.

)
