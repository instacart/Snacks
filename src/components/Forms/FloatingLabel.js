import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { colors } from '../../styles'
import withTheme from '../../styles/themer/withTheme'
import { themePropTypes } from '../../styles/themer/utils'

const styles = {
  label: {
    color: `${colors.GRAY_46}`,
    cursor: 'text',
    fontSize: '16px',
    marginTop: 0,
    marginRight: 0,
    marginBottom: 0,
    marginLeft: 0,
    pointerEvents: 'auto',
    position: 'absolute',
    transform: 'scale(1) translate(8px, 17px)',
    transformOrigin: 'left top',
    transition: 'all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
    userSelect: 'none',
    zIndex: 1,
  },
  float: {
    cursor: 'inherit',
    transform: 'scale(0.85) translate(8px, 6px)',
    pointerEvents: 'none',
  },
  error: {
    color: colors.RED_700,
  },
  disabled: {
    cursor: 'not-allowed',
    color: colors.GRAY_74,
  },
}
const getSnackStyles = snacksTheme => {
  const { action } = snacksTheme.colors

  return {
    active: {
      color: action,
    },
  }
}

@withTheme
class FloatingLabel extends Component {
  static propTypes = {
    /** Disabled styling for the label */
    disabled: PropTypes.bool,
    /** Float the label */
    float: PropTypes.bool,
    /** Show error styling */
    hasError: PropTypes.bool,
    /** HTML for attribute */
    htmlFor: PropTypes.string,
    /** Is the input in an active state */
    isActive: PropTypes.bool,
    /**
     * Optional style overrides merged into emotion css
     *
     * @deprecated
     * This prop exists for backwards compatibility and will be
     * removed in a future version
     */
    style: PropTypes.object,
    className: PropTypes.string,
    /** Label text */
    text: PropTypes.string,
    /** Snacks theme attributes provided by `Themer` */
    snacksTheme: themePropTypes,
  }

  static defaultProps = {
    disabled: false,
  }

  render() {
    const {
      className,
      float,
      disabled,
      hasError,
      htmlFor,
      isActive,
      style,
      text,
      snacksTheme,
    } = this.props

    const snacksStyles = getSnackStyles(snacksTheme)

    const inputStyles = [
      styles.label,
      float && styles.float,
      disabled && styles.disabled,
      isActive && snacksStyles.active,
      hasError && styles.error,
      style,
    ]

    return (
      <label htmlFor={htmlFor} css={inputStyles} className={className}>
        {text}
      </label>
    )
  }
}

export default FloatingLabel
