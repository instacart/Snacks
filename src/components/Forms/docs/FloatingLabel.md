The floating label must be used inside of a TextField or component that is 'position: relative'. See TextField to see how FloatingLabel is used.
