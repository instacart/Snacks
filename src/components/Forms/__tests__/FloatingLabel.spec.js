import React from 'react'
import renderer from 'react-test-renderer'
import FloatingLabel from '../FloatingLabel'

it('renders FloatingLabel correctly when floated', () => {
  const tree = renderer
    .create(
      <div>
        <FloatingLabel text="Label Text" float />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders FloatingLabel correctly when not floating', () => {
  const tree = renderer
    .create(
      <div>
        <FloatingLabel text="Label Text" />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders FloatingLabel correctly when isActive and float', () => {
  const tree = renderer
    .create(
      <div>
        <FloatingLabel text="Label Text" isActive float />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders FloatingLabel correctly when disabled and float', () => {
  const tree = renderer
    .create(
      <div>
        <FloatingLabel text="Label Text" disabled float />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders FloatingLabel correctly when disabled', () => {
  const tree = renderer
    .create(
      <div>
        <FloatingLabel text="Label Text" disabled />
      </div>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
