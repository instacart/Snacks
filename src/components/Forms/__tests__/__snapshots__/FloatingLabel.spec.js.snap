// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FloatingLabel correctly when disabled 1`] = `
.emotion-0 {
  color: #757575;
  cursor: text;
  font-size: 16px;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  pointer-events: auto;
  position: absolute;
  -webkit-transform: scale(1) translate(8px,17px);
  -ms-transform: scale(1) translate(8px,17px);
  transform: scale(1) translate(8px,17px);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 1;
  cursor: not-allowed;
  color: #BDBDBD;
}

<div>
  <label
    className="emotion-0"
  >
    Label Text
  </label>
</div>
`;

exports[`renders FloatingLabel correctly when disabled and float 1`] = `
.emotion-0 {
  color: #757575;
  cursor: text;
  font-size: 16px;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  pointer-events: auto;
  position: absolute;
  -webkit-transform: scale(1) translate(8px,17px);
  -ms-transform: scale(1) translate(8px,17px);
  transform: scale(1) translate(8px,17px);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 1;
  cursor: inherit;
  -webkit-transform: scale(0.85) translate(8px,6px);
  -ms-transform: scale(0.85) translate(8px,6px);
  transform: scale(0.85) translate(8px,6px);
  pointer-events: none;
  cursor: not-allowed;
  color: #BDBDBD;
}

<div>
  <label
    className="emotion-0"
  >
    Label Text
  </label>
</div>
`;

exports[`renders FloatingLabel correctly when floated 1`] = `
.emotion-0 {
  color: #757575;
  cursor: text;
  font-size: 16px;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  pointer-events: auto;
  position: absolute;
  -webkit-transform: scale(1) translate(8px,17px);
  -ms-transform: scale(1) translate(8px,17px);
  transform: scale(1) translate(8px,17px);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 1;
  cursor: inherit;
  -webkit-transform: scale(0.85) translate(8px,6px);
  -ms-transform: scale(0.85) translate(8px,6px);
  transform: scale(0.85) translate(8px,6px);
  pointer-events: none;
}

<div>
  <label
    className="emotion-0"
  >
    Label Text
  </label>
</div>
`;

exports[`renders FloatingLabel correctly when isActive and float 1`] = `
.emotion-0 {
  color: #757575;
  cursor: text;
  font-size: 16px;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  pointer-events: auto;
  position: absolute;
  -webkit-transform: scale(1) translate(8px,17px);
  -ms-transform: scale(1) translate(8px,17px);
  transform: scale(1) translate(8px,17px);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 1;
  cursor: inherit;
  -webkit-transform: scale(0.85) translate(8px,6px);
  -ms-transform: scale(0.85) translate(8px,6px);
  transform: scale(0.85) translate(8px,6px);
  pointer-events: none;
  color: #43B02A;
}

<div>
  <label
    className="emotion-0"
  >
    Label Text
  </label>
</div>
`;

exports[`renders FloatingLabel correctly when not floating 1`] = `
.emotion-0 {
  color: #757575;
  cursor: text;
  font-size: 16px;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  pointer-events: auto;
  position: absolute;
  -webkit-transform: scale(1) translate(8px,17px);
  -ms-transform: scale(1) translate(8px,17px);
  transform: scale(1) translate(8px,17px);
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 450ms cubic-bezier(0.23,1,0.32,1) 0ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 1;
}

<div>
  <label
    className="emotion-0"
  >
    Label Text
  </label>
</div>
`;
